Shape_detection/include/CGAL/Shape_detection/Efficient_RANSAC/Efficient_RANSAC.h

```diff
@@ -480,8 +480,11 @@ class Efficient_RANSAC {
         return false;
     }
 
-    if (callback && !callback(0.))
+    if (callback && !callback(0.)) {
+      clear_octrees();
+      clear_shape_factories();
```

Comment:
`m_num_available_points -= num_invalid;` is not needed here?

Reply:
No, I guess, because `m_num_available_points` will be initialized only after this `if`. So all early quits do not affect it.